internal/plugin/files.go

```diff
@@ -1,19 +1,35 @@
 package plugin
 
 import (
+	"fmt"
+	"hash/fnv"
+	"os"
+	"path/filepath"
 	"regexp"
+	"strconv"
 	"strings"
 
 	"github.com/pkg/errors"
+	"go.jetpack.io/devbox/internal/xdg"
 	"go.jetpack.io/devbox/plugins"
 )
 
-func getConfigIfAny(pkg, projectDir string) (*config, error) {
+func getConfigIfAny(pkg, projectDir, xdgRuntimePath string) (*config, error) {
```

Comment:
instead of passing `xdgRuntimePath` here, what do you think of the following API:

```golang
plugin.VirtenvSymlinkPath(projectDir string) string
plugin.CreateVirtenvSymlink(projectDir string) string
```

* `CreateVirtenvSymlink` calls `VirtenvSymlinkPath` to get path it needs to create.
* `CreateFilesAndShowReadme` calls `CreateVirtenvSymlink` to create the symlink if needed
* `getConfigIfAny` calls  `VirtenvSymlinkPath` to populate `virtenv` variable.

Reply:
I like it +1